src/op_mode/dhcp.py

```diff
@@ -113,7 +113,7 @@ def _get_raw_server_leases(family='inet', pool=None, sorted=None, state=[], orig
         data_lease['origin'] = 'local' # TODO: Determine remote in HA
         data_lease['hostname'] = lease.get('hostname', '-')
         # remove trailing dot to ensure consistency for `vyos-hostsd-client`
-        if data_lease['hostname'][-1] == '.':
+        if len(data_lease['hostname']) > 1 and data_lease['hostname']:
             data_lease['hostname'] = data_lease['hostname'][:-1]
```

Comment:
```suggestion
        if data_lease['hostname']:
            data_lease['hostname'] = data_lease['hostname'].rstrip('.')
```
This will avoid having the last character of hostname from being blindly removed.

Reply:
As it stands, this is breaking my script that pulls up leases from the kea database.

Reply:
ah whoops. I broke the if statement when I changed it into two commits.